Makefile

```
@@ -62,41 +91,57 @@ _api.clean.rpcmanager:
 api/rpcmanager.proto: go.sum go.mod
 	go run github.com/gfanton/grpcutil/rpcmanager/protofile > $@
 src/api/rpcmanager.pb.js: api/rpcmanager.proto
-	$(pbjs) -t json-module -w es6 -o $@ $<
+	$(pbjs) -t json-module -w commonjs -o $@ $<
 src/api/rpcmanager.pb.d.ts: api/rpcmanager.proto
 	$(pbjs) -t static-module $< | $(pbts) -o $@ -
 
 .PHONY: api.generate _api.generate.rpcmanager _api.generate.protocol
 
-## go bind
+# - Bind : Handle gomobile bind
 
-bind.init: $(TMPDIR)/.tool-versions .cache/bind/gomobile
-.cache/bind/gomobile: go.sum go.mod
+# - Bind - initialization
+bind_init_files := $(TMPDIR)/.tool-versions $(gobind) $(gomobile)
+
+$(gobind): go.sum go.mod
+	@mkdir -p $(dir $@)
+	go build -o $@ golang.org/x/mobile/cmd/gobind && chmod +x $@
+
+$(gomobile): $(gobind) go.sum go.mod
 	@mkdir -p $(dir $@)
 	go build -o $@ golang.org/x/mobile/cmd/gomobile && chmod +x $@
-	$@ init
-# FIXME(gfanton): find a more elegant way to make asdf works in the tmp directory
+	$(gomobile) init || (rm -f $@ && exit 1) # in case of failure, remove gomobile so we can init again
+
 $(TMPDIR)/.tool-versions: .tool-versions
-	@echo "> copying current `.tool-versions` in '$(TMPDIR)' folder in order to make asdf works"
+	@echo "> copying current '.tool-versions' in '$(TMPDIR)' folder in order to make asdf works"
 	@echo "> this hack is needed in order for gomobile (who is building from '$(TMPDIR)') bind to use the correct javac and go version"
 	@cp -v $< $@
-bind.clean: _bind.clean.framework
-	rm -f $(TMPDIR)/.tool-versions
-	rm -rf .cache/bind
 
-# use `nowatchdog` tags to build, see https://github.com/libp2p/go-libp2p-connmgr/issues/98
-go_files := $(shell find . -iname '*.go')
-go_deps := go.mod go.sum $(go_files)
-$(OUTPUT_FRAMEWORK): bind.init $(go_deps)
+.PHONY: bind.init
+
+# - Bind - ios framework
+
+$(weshnetcore_xcframework): $(bind_init_files) $(go_deps)
 	@mkdir -p $(dir $@)
-	.cache/bind/gomobile bind -v \
+    # need to use `nowatchdog` tags, see https://github.com/libp2p/go-libp2p-connmgr/issues/98
+	$(gomobile) bind -v \
 		-tags 'nowatchdog' -prefix=Weshnet \
-		-o $@ -target ios $(make_dir)/framework/core
-_bind.clean.framework:
-	rm -rf $(OUTPUT_FRAMEWORK)
+		-o $@ -target ios ./framework/core
```

```suggestion
		-o $@ -target ios -androidapi 19 ./framework/core
```
I wasn't able to build the android package until I set the `androidapi` version for gomobile.
It seems gomobile do some assumptions about our local environment.

---

this doesn't have any effect since you are setting this for an ios target. but even for android i dont have any problem.
indeed gomobile need to make some assumptions, if not set, he will try to get your ANDROID_HOME for a default location here: https://github.com/golang/mobile/blob/3c911d8e3edae4ddeea7f8a6e3ea9eae74f1a81a/internal/sdkpath/sdkpath.go#L34
then fetch the latest api version from the platform folder
here: https://github.com/golang/mobile/blob/3c911d8e3edae4ddeea7f8a6e3ea9eae74f1a81a/internal/sdkpath/sdkpath.go#L51

---

Sorry, this is the wrong place. Check the bellow comment, please.